openstack_terraforming/networking.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2018 The Terraformer Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openstack_terraforming
+
+import (
+	"github.com/GoogleCloudPlatform/terraformer/terraform_utils"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/security/groups"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/security/rules"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+type NetworkingGenerator struct {
+	OpenStackService
+}
+
+// createResources iterate on all openstack_networking_secgroup_v2
+func (g *NetworkingGenerator) createSecgroupResources(list *pagination.Pager) []terraform_utils.Resource {
+	resources := []terraform_utils.Resource{}
+
+	list.EachPage(func(page pagination.Page) (bool, error) {
+		groups, err := groups.ExtractGroups(page)
+		if err != nil {
+			return false, err
+		}
+
+		for _, grp := range groups {
+			resource := terraform_utils.NewResource(
+				grp.ID,
+				grp.Name,
+				"openstack_networking_secgroup_v2",
+				"openstack",
+				map[string]string{
+					"name": grp.Name,
+				},
+				[]string{},
+				map[string]string{},
+			)
+			resources = append(resources, resource)
+			resources = append(resources, g.createSecgroupRuleResources(grp.Rules)...)
+		}
+
+		return true, nil
+	})
+
+	return resources
+}
+
+// createResources iterate on all openstack_networking_secgroup_v2
+func (g *NetworkingGenerator) createSecgroupRuleResources(rules []rules.SecGroupRule) []terraform_utils.Resource {
+	resources := []terraform_utils.Resource{}
+	for _, r := range rules {
+		resource := terraform_utils.NewResource(
+			r.ID,
+			r.ID,
+			"openstack_networking_secgroup_rule_v2",
+			"openstack",
+			map[string]string{
+				"id": r.ID,
```

Comment:
same, not needed, only ID
https://github.com/terraform-providers/terraform-provider-openstack/blob/master/openstack/resource_openstack_networking_secgroup_rule_v2.go#L180